shadow.c

```diff
@@ -549,15 +559,29 @@ static char *get_full_path(const char *filename TSRMLS_DC)
 		SHADOW_G(curdir) = getcwd(NULL, 0);
 	}
 
+#if PHP_VERSION_ID < 50600
 	new_state.cwd = strdup(SHADOW_G(curdir));
+#else
+	new_state.cwd = estrdup(SHADOW_G(curdir));
```

Comment:
@alex-vlasov if this is the case, could you add some comments referring to the commit or changelog entry between PHP 5.5 and 5.6 which would explain the need for this branching? It will be useful for further understanding of the logic.